contracts/utils/ExternalMulticallSimulator.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "../utils/SelfMulticall.sol";
+import "./interfaces/IExternalMulticallSimulator.sol";
+import "../vendor/@openzeppelin/contracts@4.9.5/utils/Address.sol";
+
+/// @title Contract that simulates external calls in single or batched form
+/// @notice The contract only allows eth_call to be used for while making
+/// external calls to ensure that it is only used for simulating outcomes,
+/// rather than sending actual transactions
+contract ExternalMulticallSimulator is
+    SelfMulticall,
+    IExternalMulticallSimulator
+{
+    /// @dev Only allows calls where the sender is address-zero and the gas
+    /// price is zero, which indicates that the call is made through eth_call
+    modifier onlyEthCall() {
```

Comment:
Nit: I might have called it `onlyVoidSignerEthCall()` instead

Reply:
I think I should move the modifier in the function instead considering it's only used once, then we wouldn't need to name it too

Reply:
Done in https://github.com/api3dao/contracts/pull/126/commits/1962d79d46e463b9083d9b2b63b9de6779d48cd2